test/hotspot/jtreg/serviceability/dcmd/compiler/PerfMapTest.java

```diff
@@ -113,4 +113,14 @@ public void specifiedDefaultMapFile() throws IOException {
         run(new JMXExecutor(), "Compiler.perfmap " + path.toString(), path);
         Files.deleteIfExists(path);
     }
+
+    @Test
+    public void logErrorsDcmdOutputStream() throws IOException {
+        String test_dir = System.getProperty("test.dir", ".");
+        Path path = Paths.get("nonexistent", test_dir);
+        OutputAnalyzer output = new JMXExecutor().execute("Compiler.perfmap %s".formatted(path));
+        output.shouldContain("Failed to create nonexistent/%s for perf map".formatted(test_dir));
+        output.shouldNotHaveExitValue(0);
+        Files.deleteIfExists(path);
```

Comment:
If the file exists, that means the expected error message will not be found, which means an exception will be thrown before you get to the `Files.deleteIfExits(path)` call. If the file doesn't exist, then there is nothing to delete. So as things stand now this call will never delete anything. Maybe put it in a finally block so if the file does exist it will get deleted.

Reply:
Makes sense, I added the finally block.